src/components/AppliedGoalParentFilter/AppliedGoalParentFilter.tsx

```diff
@@ -6,14 +9,41 @@ interface AppliedGoalParentFilterProps {
     label?: string;
     value?: string[];
     readOnly?: boolean;
+    onChange?: (value?: string[]) => void;
+    onClose?: () => void;
 }
 
-export const AppliedGoalParentFilter = ({ label, value = [], readOnly }: AppliedGoalParentFilterProps) => {
+export const AppliedGoalParentFilter = ({
+    label,
+    value = [],
+    readOnly,
+    onChange,
+    onClose,
+}: AppliedGoalParentFilterProps) => {
     const { data: projects } = trpc.project.getByIds.useQuery({ ids: value });
 
+    const handleChange = useCallback(
+        (values: { id: string }[]) => {
+            onChange?.(values.map(({ id }) => id));
+        },
+        [onChange],
+    );
+
+    const onClearFilter = useCallback(() => {
+        onChange?.();
+        onClose?.();
+    }, [onChange, onClose]);
```

Comment:
you could probably wrap these two into a single hook and use once instead of copypaste